File: 5차 과제/6-3/py 소스 파일/6_3.py
from header6_3 import queue
from func6_3_1 import Linked_queue

def main():
    queue=Linked_queue()
    print("*****연결 큐 연산*****")
    queue.enqueue('A')
    print()

    queue.enqueue('B')
    print()

    queue.enqueue('C')
    print(f"peek item:{queue.front.data}\n")

    item=queue.dequeue()
    print(f"\t삭제 데이터:{item}")

    item=queue.dequeue()
    print(f"\t삭제 데이터:{item}")

    item=queue.dequeue()
    print(f"\t\t삭제 데이터:{item}")

    queue.enqueue('D')
    print()

    queue.enqueue('E')
    print()

if __name__ == "__main__":
    main()

File: 5차 과제/6-3/C 소스 파일/6-3.c
#include <stdio.h>
#include "6-3.h"

queue* front = NULL;
queue* rear = NULL;

int main() {
    printf("*****연결 큐 연산*****\n");
    enqueue('A');
    printf("\n");

    enqueue('B');
    printf("\n");

    enqueue('C');
    printf("peek item: %c\n\n", front->data);

    element2 item = dequeue();
    printf("\t삭제 데이터: %c\n", item);

    item = dequeue();
    printf("\t삭제 데이터: %c\n", item);

    item = dequeue();
    printf("\t삭제 데이터: %c\n", item);

    enqueue('D');
    printf("\n");

    enqueue('E');
    printf("\n");
    return 0;
}
